bootstrap/internal/controllers/rke2config_controller.go

```diff
@@ -341,6 +372,39 @@ func (r *RKE2ConfigReconciler) handleClusterNotInitialized(ctx context.Context,
 	return ctrl.Result{}, nil
 }
 
+func (r *RKE2ConfigReconciler) generateFileListIncludingRegistries(scope *Scope, ctx context.Context, configFiles []bootstrapv1.File) ([]bootstrapv1.File, error) {
+	registries, registryFiles, err := rke2.GenerateRegistries(rke2.RKE2ConfigRegistry{
+		Registry: scope.Config.Spec.PrivateRegistriesConfig,
+		Client:   r.Client,
+		Ctx:      ctx,
+		Logger:   scope.Logger,
+	})
+
+	if err != nil {
+		scope.Logger.Error(err, "unable to generate registries.yaml for Init Control Plane node")
+		return nil, err
+	}
+
+	registriesYAML, err := kubeyaml.Marshal(registries)
+	if err != nil {
+		scope.Logger.Error(err, "unable to marshall registries.yaml")
+		return nil, err
+	}
+	scope.Logger.Info("Registries.yaml marshalled successfully")
+
+	initRegistriesFile := bootstrapv1.File{
+		Path:        rke2.DefaultRKE2RegistriesLocation,
+		Content:     string(registriesYAML),
+		Owner:       "root:root",
```

Comment:
We can use the constants defined above for these 2.

Reply:
Done, thank you!